app/views/dashboard/index.html.erb

```diff
@@ -11,15 +11,20 @@
     <div id='dashboard' class="gridster col-md-12">
       <ul>
         <% User.current.widgets.each do |widget| %>
-          <%= content_tag(:li, widget_data(widget)) do %>
-            <div class='widget_control'>
-              <a class='remove' data-url='<%= widget_path(widget) %>'>&times;</a>
-              <a class='minimize'>&minus;</a>
-            </div>
-            <div class="widget <%= widget.name.parameterize %>">
-              <%= render_widget(widget) %>
-            </div>
-          <% end %>
+            <%= content_tag(:li, widget_data(widget)) do %>
+              <div class='widget_control'>
+                <a class='remove' data-url='<%= widget_path(widget) %>'>&times;</a>
+                <a class='minimize'>&minus;</a>
+              </div>
+              <div class="widget <%= widget.name.parameterize %>">
+
+                <%= begin
+                      render_widget(widget)
+                    rescue ActionView::MissingTemplate => e; e.class
```

Comment:
Possibly, but I liked this, because it was

   1. less work
   2. letting the user to fix their dashboard: if they already have customized dashboard, I though it would be friendlier for the user just reporting the issue to them.

On the other hand, that doesn't solve issues, when the dashboard renames the template: in that case we should not save the paths to the database, but rather load that from the widget defitntion, witch would identify the problem.